test/integration/scheduler/predicates_test.go

```diff
@@ -920,3 +926,165 @@ func TestNodePIDPressure(t *testing.T) {
 
 	cleanupPods(cs, t, []*v1.Pod{testPod})
 }
+
+// TestEvenPodsSpreadPredicate verifies that EvenPodsSpread predicate functions well.
+func TestEvenPodsSpreadPredicate(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EvenPodsSpread, true)()
+	// Apply feature gates to enable EvenPodsSpread
+	defer algorithmprovider.ApplyFeatureGates()()
+
+	context := initTest(t, "eps-predicate")
+	cs := context.clientSet
+	ns := context.ns.Name
+	defer cleanupTest(t, context)
+	// Add 4 nodes.
+	nodes, err := createNodes(cs, "node", nil, 4)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	for i, node := range nodes {
+		// Apply labels "zone: zone-{0,1}" and "node: <node name>" to each node.
+		labels := map[string]string{
+			"zone": fmt.Sprintf("zone-%d", i/2),
+			"node": node.Name,
+		}
+		if err = testutils.AddLabelsToNode(cs, node.Name, labels); err != nil {
+			t.Fatalf("Cannot add labels to node: %v", err)
+		}
+		if err = waitForNodeLabels(cs, node.Name, labels); err != nil {
+			t.Fatalf("Adding labels to node failed: %v", err)
+		}
+	}
+
+	pause := imageutils.GetPauseImageName()
+	tests := []struct {
+		name string
+		pod  *v1.Pod
+		pods []*v1.Pod
+		fits bool
```

Comment:
nit: add a comment on what does `fits` mean. `success` is probably a better name.